query-engine/core/src/query_ast/write.rs

```diff
@@ -64,7 +64,13 @@ impl WriteQuery {
         match self {
             Self::CreateRecord(_) => returns_id,
             Self::CreateManyRecords(_) => false,
-            Self::UpdateRecord(_) => returns_id,
+            Self::UpdateRecord(UpdateRecord::WithExplicitSelection(ur)) => {
+                ur.selected_fields.is_superset_of(field_selection)
+            }
+            Self::UpdateRecord(UpdateRecord::WithImplicitSelection(ur)) => {
+                ur.selected_fields().is_superset_of(field_selection)
+            }
+            Self::UpdateRecord(UpdateRecord::WithoutSelection(_)) => returns_id,
```

Comment:
This is a bit out of scope for this PR.

This method, `WriteQuery::returns`, is used to compute reload nodes by checking whether a node's field selection fulfills its children data dependencies. It had not been updated after making `UpdateRecord` nodes return an arbitrary selection set. This means that some reload nodes might've not been added in cases it's needed.